diff -ur xkobo-1.11/enemies.h xkobo-1.11-p1/enemies.h
--- xkobo-1.11/enemies.h	1996-04-04 06:58:50.000000000 +0900
+++ xkobo-1.11-p1/enemies.h	2011-01-30 20:21:37.979485783 +0900
@@ -62,7 +62,7 @@
 extern const enemy_kind cannon;
 extern const enemy_kind pipe1;
 extern const enemy_kind core;
-extern const enemy_kind pipe2;
+extern const enemy_kind pipe_2;
 extern const enemy_kind rock;
 extern const enemy_kind ring;
 extern const enemy_kind enemy_m1;
@@ -209,10 +209,10 @@
         state = notuse;
     }
     else if (ek == &core){
-        enemies.make(&pipe2, x>>SHIFT, y>>SHIFT, 0, 0, 3);
-        enemies.make(&pipe2, x>>SHIFT, y>>SHIFT, 0, 0, 7);
-        enemies.make(&pipe2, x>>SHIFT, y>>SHIFT, 0, 0, 1);
-        enemies.make(&pipe2, x>>SHIFT, y>>SHIFT, 0, 0, 5);
+        enemies.make(&pipe_2, x>>SHIFT, y>>SHIFT, 0, 0, 3);
+        enemies.make(&pipe_2, x>>SHIFT, y>>SHIFT, 0, 0, 7);
+        enemies.make(&pipe_2, x>>SHIFT, y>>SHIFT, 0, 0, 1);
+        enemies.make(&pipe_2, x>>SHIFT, y>>SHIFT, 0, 0, 5);
         enemies.make(&explosion, x>>SHIFT, y>>SHIFT);
         state = notuse;
         manage.destroyed_a_core();
@@ -304,7 +304,7 @@
 
 inline int _enemy::is_pipe()
 {
-    return ((state != notuse) && ((ek == &pipe1) || (ek == &pipe2)));
+    return ((state != notuse) && ((ek == &pipe1) || (ek == &pipe_2)));
 }
 
 
diff -ur xkobo-1.11/enemy.C xkobo-1.11-p1/enemy.C
--- xkobo-1.11/enemy.C	1996-03-14 01:34:27.000000000 +0900
+++ xkobo-1.11-p1/enemy.C	2011-01-30 20:22:13.573051971 +0900
@@ -612,17 +612,17 @@
     p ^= a;
     int a=0, b=0, c=0, d=0;
     if (p & U_MASK)
-      a=enemies.make(&pipe2, (x>>SHIFT), (y>>SHIFT), 0, 0, 1);
+      a=enemies.make(&pipe_2, (x>>SHIFT), (y>>SHIFT), 0, 0, 1);
     if (p & R_MASK)
-      b=enemies.make(&pipe2, (x>>SHIFT), (y>>SHIFT), 0, 0, 3);
+      b=enemies.make(&pipe_2, (x>>SHIFT), (y>>SHIFT), 0, 0, 3);
     if (p & D_MASK)
-      c=enemies.make(&pipe2, (x>>SHIFT), (y>>SHIFT), 0, 0, 5);
+      c=enemies.make(&pipe_2, (x>>SHIFT), (y>>SHIFT), 0, 0, 5);
     if (p & L_MASK)
-      d=enemies.make(&pipe2, (x>>SHIFT), (y>>SHIFT), 0, 0, 7);
+      d=enemies.make(&pipe_2, (x>>SHIFT), (y>>SHIFT), 0, 0, 7);
     manage.add_score(10);
     state = notuse;
 }
-const enemy_kind pipe2 = {
+const enemy_kind pipe_2 = {
     0,
     &_enemy::make_pipe2,
     &_enemy::move_pipe2,
diff -ur xkobo-1.11/file.C xkobo-1.11-p1/file.C
--- xkobo-1.11/file.C	1997-01-10 11:17:36.000000000 +0900
+++ xkobo-1.11-p1/file.C	2011-01-30 20:20:05.495532995 +0900
@@ -29,9 +29,10 @@
 #include <errno.h>
 }
 
-int load_block(char *file_name, void *t, int len) 
+int load_block(char *file_name, void *_t, int len) 
 {
     int i, fd, ret = 0;
+    char *t=(char*)_t;
     
     fd = open(file_name, O_RDONLY, 0);
     if (fd == -1) return -1;
@@ -55,9 +56,10 @@
     return ret;
 }
 
-int save_block(char *file_name, void *t, int len) 
+int save_block(char *file_name, void *_t, int len) 
 {
     int i, fd, ret = 0;
+    char *t=(char*)_t;
      
     fd = open(file_name, O_CREAT|O_WRONLY, 0644);
     if (fd == -1) return -1;
diff -ur xkobo-1.11/map.h xkobo-1.11-p1/map.h
--- xkobo-1.11/map.h	1996-04-04 06:58:52.000000000 +0900
+++ xkobo-1.11-p1/map.h	2011-01-30 20:13:32.329104613 +0900
@@ -38,10 +38,10 @@
 #define HIT_MASK       (CORE | U_MASK | R_MASK | D_MASK | L_MASK)
 
 class _map{
-    const int sx_log2 = MAP_SIZEX_LOG2;
-    const int sy_log2 = MAP_SIZEY_LOG2;
-    const int sx = 1 << sx_log2;
-    const int sy = 1 << sy_log2;
+    static const int sx_log2 = MAP_SIZEX_LOG2;
+    static const int sy_log2 = MAP_SIZEY_LOG2;
+    static const int sx = 1 << sx_log2;
+    static const int sy = 1 << sy_log2;
     int sitex[SITE_MAX];
     int sitey[SITE_MAX];
     int site_max;
diff -ur xkobo-1.11/xkobo.C xkobo-1.11-p1/xkobo.C
--- xkobo-1.11/xkobo.C	1996-06-15 02:27:27.000000000 +0900
+++ xkobo-1.11-p1/xkobo.C	2011-01-30 20:14:02.808837284 +0900
@@ -51,7 +51,7 @@
 
 static int      signal_delivered = 1;
 
-static void sig_handle(...)
+static void sig_handle(int)
 {
     signal_delivered = 1;
 }
