diff --git a/Bonus/goodict.cgi b/Bonus/goodict.cgi
index 8f443c1..930e15c 100755
--- a/Bonus/goodict.cgi
+++ b/Bonus/goodict.cgi
@@ -4,12 +4,18 @@
 # e:      file:/cgi-bin/goodict.cgi?%s
 # w:      file:/cgi-bin/goodict.cgi?%s
 # j:      file:/cgi-bin/goodict.cgi?%s
+#
+# e:0:word  start with word
+# e:1:word  perfect match
+# e:2:word  end with word
+# e:3:word  search body text
+# e:6:word  search title
+# e:word    perfect match
 
-use NKF;
-#$mode = 0; # substring
-$mode = 1;  # perfect match
-#$mode = 3; # search body text
-$url = "http://dictionary.goo.ne.jp";
+use Encode;
+use Encode::Guess qw/euc-jp utf8/;
+
+$url = "http://dictionary.goo.ne.jp/";
 $_ = $ENV{"QUERY_STRING"};
 if (/^e:/) {
     $kind = 'ej';
@@ -19,12 +25,18 @@ if (/^e:/) {
     $kind = 'jn';
 }
 s@^[ewjs]:@@ && s@^//@@ && s@/$@@;
+if (/^([01236]):/){
+    $mode=$1;
+    s/^[01236]://;
+}else{
+    $mode="1";
+}
 if ($_) {
 	s/\+/ /g;
 	s/%([\da-f][\da-f])/pack('C', hex($1))/egi;
-	$_ = nkf("-e", $_);
+	$_ = encode("utf8", decode("Guess", $_));
 	s/[\000-\040\+:#?&%<>"\177-\377]/sprintf('%%%02X', unpack('C', $&))/eg;
-	$url .= "/search.php?MT=$_&kind=$kind&mode=$mode";
+	$url .= "srch/$kind/$_/m$mode"."u/";
 } else {
 	$input = "w3m-control: GOTO_LINK";
 }
diff --git a/Bonus/google.cgi b/Bonus/google.cgi
index 0016340..05a18c4 100755
--- a/Bonus/google.cgi
+++ b/Bonus/google.cgi
@@ -8,7 +8,7 @@ $url = "http://www.google.com/";
 $_ = $ENV{"QUERY_STRING"};
 s@^g(oogle)?:@@ && s@^//@@ && s@/$@@;
 if ($_) {
-	s/\+/ /g;
+	s/%0C/ /g;
 	s/%([\da-f][\da-f])/pack('C', hex($1))/egi;
 	s/[\000-\040\+:#?&%<>"\177-\377]/sprintf('%%%02X', unpack('C', $&))/eg;
 	$url .= "search?q=$_&hl=ja&lr=lang_ja&ie=EUC-JP";
@@ -18,6 +18,7 @@ if ($_) {
 print <<EOF;
 w3m-control: GOTO $url
 w3m-control: DELETE_PREVBUF
+w3m-control: MOVE_DOWN
 w3m-control: SEARCH \\[
 w3m-control: MOVE_RIGHT
 ${input}
