diff --git a/scripts/w3mman/w3mman2html.cgi.in b/scripts/w3mman/w3mman2html.cgi.in
index 705ab7a..ad2c45e 100755
--- a/scripts/w3mman/w3mman2html.cgi.in
+++ b/scripts/w3mman/w3mman2html.cgi.in
@@ -134,6 +134,11 @@ while(<F>) {
   s@((\<b\>)?[\200-\377].(\</b\>)?)\010{1,2}__@<u>$1</u>@g;
   s@((\<b\>)?(\&\w+\;|.)(\</b\>)?)\010_@<u>$1</u>@g;
   s@.\010(.)@$1@g;
+  s@\e\[0m@</b></u>@g;
+  s@\e\[1m@<b>@g;
+  s@\e\[22m@</b>@g;
+  s@\e\[4m@<u>@g;
+  s@\e\[24m@</u>@g;
 
   s@\</b\>\</u\>\<b\>_\</b\>\<u\>\<b\>@_@g;
   s@\</u\>\<b\>_\</b\>\<u\>@_@g;
