diff -ur scilab-5.3.3/configure build32/configure
--- scilab-5.3.3/configure	2011-07-20 17:15:42.000000000 +0900
+++ build32/configure	2012-07-01 13:40:26.629099741 +0900
@@ -13476,7 +13476,7 @@
     saved_ac_java_classpath=$ac_java_classpath
     DEFAULT_JAR_DIR="/usr/share/java/ /usr/lib/java/ /usr/share/java /usr/share/java/jar /opt/java/lib /usr/local/java /usr/local/java/jar /usr/local/share/java /usr/local/share/java/jar /usr/local/lib/java $(ls -d /usr/share/java/*/ 2>/dev/null) $(ls -d /usr/lib64/*/ 2>/dev/null) $(ls -d /usr/lib/*/ 2>/dev/null)  $(ls -d /usr/share/*/lib/ 2>/dev/null)"
     for jardir in "`pwd`/thirdparty" "`pwd`/jar" $DEFAULT_JAR_DIR "$_user_libdir"; do
-      for jar in "$jardir/jrosetta-API.jar" "$jardir/libjrosetta-API.jar" "$jardir/libjrosetta-API-java.jar" "$jardir/jrosetta-API*.jar"; do
+      for jar in "$jardir/jrosetta-API.jar" "$jardir/libjrosetta-API.jar" "$jardir/libjrosetta-API-java.jar" "$jardir/jrosetta-API*.jar" "$jardir/jrosetta-api*.jar"; do
 #    jar=`echo $jar|sed -e 's/ /\\ /'`
 #    echo "protected $jar"
 #    jar_resolved=`ls $jar 2>/dev/null`
