diff -ur VTK6.0.0/IO/ParallelNetCDF/Testing/Cxx/CMakeLists.txt build/IO/ParallelNetCDF/Testing/Cxx/CMakeLists.txt
--- VTK6.0.0/IO/ParallelNetCDF/Testing/Cxx/CMakeLists.txt	2013-06-13 04:47:10.000000000 +0900
+++ build/IO/ParallelNetCDF/Testing/Cxx/CMakeLists.txt	2013-08-03 14:38:03.110845470 +0900
@@ -5,6 +5,7 @@
 )
 
 vtk_module_test_executable(${vtk-module}CxxTests ${Tests})
+target_link_libraries(${vtk-module}CxxTests vtkInteractionStyle)
 
 if(VTK_LARGE_DATA_ROOT)
   add_test(NAME vtkIOParallelNetCDFCxx-TestPNetCDFPOPReader
