diff -ur VTK-6.2.0-orig/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfArray.h VTK-6.2.0/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfArray.h
--- VTK-6.2.0-orig/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfArray.h	2015-03-04 05:37:14.000000000 +0900
+++ VTK-6.2.0/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfArray.h	2015-05-17 18:21:54.457414931 +0900
@@ -30,9 +30,6 @@
 #define XDMF_LONGEST_LENGTH     (XdmfInt64)~0
 #define XDMF_ARRAY_TAG_LENGTH  80
 
-namespace xdmf2
-{
-
 //! Self Describing Data Structure
 /*!
 This is the SuperClass for All XDMF Arrays.
@@ -51,6 +48,9 @@
 #include <stdlib.h>
 #endif
 
+namespace xdmf2
+{
+
 class XDMF_EXPORT XdmfArray : public XdmfDataDesc {
 
 public:
diff -ur VTK-6.2.0-orig/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt VTK-6.2.0/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt
--- VTK-6.2.0-orig/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt	2015-03-04 05:37:14.000000000 +0900
+++ VTK-6.2.0/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt	2015-05-17 18:16:37.463398799 +0900
@@ -154,7 +154,7 @@
 add_library(XdmfCore ${LIBTYPE} ${XdmfCoreSources})
 link_directories(${XDMF_LIBRARY_DIRS})
 
-target_link_libraries(XdmfCore ${XDMF_LIBRARIES} ${vtkxdmf3_LINK_DEPENDS})
+target_link_libraries(XdmfCore ${XDMF_LIBRARIES})
 vtk_target_install(XdmfCore)
 
 if(WIN32)
