--- mpich2-1.4.1p1/Makefile.in	2011-09-02 03:55:44.000000000 +0900
+++ build32/Makefile.in	2012-05-27 03:45:33.821468648 +0900
@@ -202,16 +202,16 @@
 	if [ -s $(srcdir)/doc/installguide/install.pdf ] ; then $(INSTALL_DATA)  $(srcdir)/doc/installguide/install.pdf ${DESTDIR}${docdir}/install.pdf ; fi
 	if [ -s $(srcdir)/doc/smpd/smpd_pmi.pdf ] ; then $(INSTALL_DATA)  $(srcdir)/doc/smpd/smpd_pmi.pdf ${DESTDIR}${docdir}/smpd_pmi.pdf ; fi
 	if [ -s $(srcdir)/doc/logging/logging.pdf ] ; then $(INSTALL_DATA)  $(srcdir)/doc/logging/logging.pdf ${DESTDIR}${docdir}/logging.pdf ; fi
-	(cd src/env && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) prefix=${DESTDIR}${prefix} exec_prefix=${DESTDIR}${exec_prefix} bindir=${DESTDIR}${bindir} libdir=${DESTDIR}${libdir} includedir=${DESTDIR}${includedir} install)
-	(cd src/packaging/pkgconfig && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) prefix=${DESTDIR}${prefix} exec_prefix=${DESTDIR}${exec_prefix} bindir=${DESTDIR}${bindir} libdir=${DESTDIR}${libdir} includedir=${DESTDIR}${includedir} install)
-	(cd src/packaging/envmods && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) prefix=${DESTDIR}${prefix} exec_prefix=${DESTDIR}${exec_prefix} bindir=${DESTDIR}${bindir} libdir=${DESTDIR}${libdir} includedir=${DESTDIR}${includedir} install)
+	(cd src/env && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) DESTDIR=${DESTDIR} prefix=${prefix} exec_prefix=${exec_prefix} bindir=${bindir} libdir=${libdir} includedir=${includedir} install)
+	(cd src/packaging/pkgconfig && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) DESTDIR=${DESTDIR} prefix=${prefix} exec_prefix=${exec_prefix} bindir=${bindir} libdir=${libdir} includedir=${includedir} install)
+	(cd src/packaging/envmods && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) DESTDIR=${DESTDIR} prefix=${prefix} exec_prefix=${exec_prefix} bindir=${bindir} libdir=${libdir} includedir=${includedir} install)
 	for dir in @bindings_dirs@ - ; do \
 		if [ "$$dir" = "-" ] ; then break ; fi ;\
-		(cd $$dir && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) prefix=${DESTDIR}${prefix} exec_prefix=${DESTDIR}${exec_prefix} bindir=${DESTDIR}${bindir} libdir=${DESTDIR}${libdir} includedir=${DESTDIR}${includedir} install) ;\
+		(cd $$dir && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) DESTDIR=${DESTDIR} prefix=${prefix} exec_prefix=${exec_prefix} bindir=${bindir} libdir=${libdir} includedir=${includedir} install) ;\
 	done
 	for dir in @other_install_dirs@ - ; do \
 		if [ "$$dir" = "-" ] ; then break ; fi ;\
-		(cd $$dir && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) prefix=${DESTDIR}${prefix} exec_prefix=${DESTDIR}${exec_prefix} bindir=${DESTDIR}${bindir} libdir=${DESTDIR}${libdir} includedir=${DESTDIR}${includedir} install) ;\
+		(cd $$dir && $(MAKE) INSTALL_STRIP_FLAG=$(INSTALL_STRIP_FLAG) PACKAGE=$(PACKAGE) DESTDIR=${DESTDIR} prefix=${prefix} exec_prefix=${exec_prefix} bindir=${bindir} libdir=${libdir} includedir=${includedir} install) ;\
 	done
 
 install-strip:
--- mpich2-1.4.1p1/src/env/Makefile.in	2011-09-02 03:55:45.000000000 +0900
+++ build32/src/env/Makefile.in	2012-05-27 17:23:25.220284376 +0900
@@ -135,21 +135,21 @@
 #
 # 
 install-preamble:
-	@if [ "@prefix@" != ${DESTDIR}${prefix} -o \
+	@if [ "@prefix@" != ${prefix} -o \
 	      "@includedir@" != ${includedir} -o \
 	      "@libdir@" != ${libdir} ] ; then \
 	      echo "Updating the compilation script for the new paths"; \
 	      for pg in mpicc mpicxx mpif90 mpif77 ; do \
 	         if [ ! -s $${pg} ] ; then continue ; fi ; \
-	         sed -e 's%includedir=.*%includedir='${DESTDIR}${includedir}% \
-	           -e 's%libdir=.*%libdir='${DESTDIR}${libdir}% \
-		   -e 's%prefix=.*%prefix='${DESTDIR}${prefix}% $${pg} > $${pg}.new ;\
+	         sed -e 's%includedir=.*%includedir='${includedir}% \
+	           -e 's%libdir=.*%libdir='${libdir}% \
+		   -e 's%prefix=.*%prefix='${prefix}% $${pg} > $${pg}.new ;\
 		 if [ ! -s $${pg}.orig ] ; then mv -f $${pg} $${pg}.orig ; fi;\
 		 mv -f $${pg}.new $${pg} ;\
 	         if [ ! -s $${pg}.conf ] ; then continue ; fi ; \
-	         sed -e 's%includedir=.*%includedir='${DESTDIR}${includedir}% \
-	           -e 's%libdir=.*%libdir='${DESTDIR}${libdir}% \
-		   -e 's%prefix=.*%prefix='${DESTDIR}${prefix}% \
+	         sed -e 's%includedir=.*%includedir='${includedir}% \
+	           -e 's%libdir=.*%libdir='${libdir}% \
+		   -e 's%prefix=.*%prefix='${prefix}% \
 		   $${pg}.conf > $${pg}.conf.new ;\
 		 if [ ! -s $${pg}.conf.orig ] ; then \
 		    mv -f $${pg}.conf $${pg}.conf.orig ; fi;\
